central/rbac/k8srole/datastore/datastore_impl.go

```diff
@@ -62,10 +69,49 @@ func (d *datastoreImpl) RemoveRole(ctx context.Context, id string) error {
 }
 
 func (d *datastoreImpl) Search(ctx context.Context, q *v1.Query) ([]searchPkg.Result, error) {
-	return d.searcher.Search(ctx, q)
+	return d.getSearchResults(ctx, q)
 }
 
 // Count returns the number of search results from the query
 func (d *datastoreImpl) Count(ctx context.Context, q *v1.Query) (int, error) {
-	return d.searcher.Count(ctx, q)
+	return d.getCountResults(ctx, q)
+}
+
+func (d *datastoreImpl) searchRoles(ctx context.Context, q *v1.Query) ([]*storage.K8SRole, []searchPkg.Result, error) {
```

Comment:
If SearchRaw calls GetByQuery then this method can probably be eliminated

Reply:
I will leave it to you if you want to move the functionality up to `SearchRoles` and eliminate the helper or keep the helper.  I have no real preference